memiavl/db.go

```diff
@@ -604,10 +617,12 @@ func (db *DB) Copy() *DB {
 
 func (db *DB) copy(cacheSize int) *DB {
 	mtree := db.MultiTree.Copy(cacheSize)
+
 	return &DB{
-		MultiTree: *mtree,
-		logger:    db.logger,
-		dir:       db.dir,
+		MultiTree:          *mtree,
+		logger:             db.logger,
+		dir:                db.dir,
+		snapshotWriterPool: db.snapshotWriterPool,
```

Comment:
we should make the snapshot rewriting goroutines cancellable, and close the pool when closing the db, but that's a separate issue.